### PR TITLE
Convert some function exprs to arrows

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Also formatted some files

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ The call to the reduceRight `callbackFn` would look something like
 this:
 
 ```js
-arr.reduceRight(function(accumulator, currentValue, index, array) {
+arr.reduceRight(function (accumulator, currentValue, index, array) {
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:82]:
```suggestion
arr.reduceRight((accumulator, currentValue, index, array) => {
```

File: files/en-us/web/javascript/reference/global_objects/function/bind/index.md [hunk 2]
```diff
@@ -144,11 +144,11 @@ function LateBloomer() {
 }
 
 // Declare bloom after a delay of 1 second
-LateBloomer.prototype.bloom = function() {
+LateBloomer.prototype.bloom = function () {
   window.setTimeout(this.declare.bind(this), 1000);
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/function/bind/index.md:148]: `setTimeout()` also works in workers, so no need for `window`:
```suggestion
  setTimeout(this.declare.bind(this), 1000);
```

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/apply/index.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ function object.
 The following code traps a function call.
 
 ```js
-const p = new Proxy(function() {}, {
+const p = new Proxy(function () {}, {
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/proxy/proxy/apply/index.md:69]: Why not:
```suggestion
const p = new Proxy(() => {}, {
```
?

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md [hunk 4]
```diff
@@ -62,7 +62,7 @@ If the following invariants are violated, the proxy will throw a {{jsxref("TypeE
 The following code traps the {{jsxref("Operators/new", "new")}} operator.
 
 ```js
-const p = new Proxy(function() {}, {
+const p = new Proxy(function () {}, {
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md:65]:
```suggestion
const p = new Proxy(() => {}, {
```

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md [hunk 5]
```diff
@@ -76,7 +76,7 @@ console.log(new p(1).value); // "called: 1"
 The following code violates the invariant.
 
 ```js example-bad
-const p = new Proxy(function() {}, {
+const p = new Proxy(function () {}, {
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md:79]:
```suggestion
const p = new Proxy(() => {}, {
```

File: files/en-us/web/javascript/reference/global_objects/regexp/index.md [hunk 6]
```diff
@@ -240,16 +240,16 @@ Starting with Firefox 34, in the case of a capturing group with quantifiers prev
 
 ```js
 // Firefox 33 or older
-'x'.replace(/x(.)?/g, function(m, group) {
-  console.log("'group:" + group + "'");
+'x'.replace(/x(.)?/g, (m, group) => {
+  console.log('group:', group);
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/regexp/index.md:244]:
```suggestion
  console.log(`group: ${group}`);
```

File: files/en-us/web/javascript/reference/global_objects/regexp/index.md [hunk 7]
```diff
@@ -240,16 +240,16 @@ Starting with Firefox 34, in the case of a capturing group with quantifiers prev
...
 });
-// 'group:'
+// group: ""
 
 // Firefox 34 or newer
-'x'.replace(/x(.)?/g, function(m, group) {
-  console.log("'group:" + group + "'");
+'x'.replace(/x(.)?/g, (m, group) => {
+  console.log('group:', group);
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/regexp/index.md:250]:
```suggestion
  console.log(`group: ${group}`);
```

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md [hunk 8]
```diff
@@ -103,9 +103,9 @@ would be returned without calling `callbackFn`.
 Some example run-throughs of the function would look like this:
 
 ```js
-[0, 1, 2, 3, 4].reduceRight(function(accumulator, currentValue, index, array) {
-  return accumulator + currentValue;
-});
+[0, 1, 2, 3, 4].reduceRight(
+  (accumulator, currentValue, index, array) => accumulator + currentValue,
```
Comment (OnkarRuikar) [inline files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:107]:
```suggestion
  (accumulator, currentValue) => accumulator + currentValue
```
`index` and `array` are not being used

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:107]: Yes, but this is meant to be an example. The table below gives the values for all parameters.

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md (hunk 5 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md:79]: This is the trap for `construct`; arrows can't be constructed.

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md (hunk 4 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/proxy/proxy/construct/index.md:65]: This is the trap for `construct`; arrows can't be constructed.

File: files/en-us/web/javascript/reference/global_objects/proxy/proxy/apply/index.md (hunk 3 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/proxy/proxy/apply/index.md:69]: Arrow functions don't have `this` so they are less illustrative of how trapping `apply` works.

File: files/en-us/web/javascript/reference/global_objects/regexp/index.md (hunk 6 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/regexp/index.md:244]:
```suggestion
  console.log(`group: ${JSON.stringify(group)}`);
```

**Review by teoli2003 — APPROVED** [review]

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md (hunk 8 above)
Comment (OnkarRuikar) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:107]: why the trailing `,`? 
does the linter do that? or it is to indicate there is an argument available?

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:107]: Prettier will add a trailing comma to all arguments. It's a good industry practice.

**Pull request merged**